napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,510 @@
+
+    if bevel:
+        bevel_count = len(normals)
+        if not closed:
+            bevel_count -= 2
+    else:
+        for i in range(1, len(normals) - 1):
+            cos_angle = (
+                normals[i - 1, 0] * normals[i, 0]
+                + normals[i - 1, 1] * normals[i, 1]
+            )
+            if cos_limit > cos_angle:
+                # bevel
+                bevel_count += 1
+
+        if closed:
+            cos_angle = (
+                normals[-2, 0] * normals[-1, 0]
+                + normals[-2, 1] * normals[-1, 1]
+            )
+            if cos_limit > cos_angle:
+                bevel_count += 1
+            cos_angle = (
+                normals[-1, 0] * normals[0, 0] + normals[-1, 1] * normals[0, 1]
+            )
+            if cos_limit > cos_angle:
+                bevel_count += 1
+
+    return point_count + bevel_count
+
+
+@njit(cache=True, inline='always')
+def _set_centers_and_offsets(
+    centers: np.ndarray,
+    offsets: np.ndarray,
+    triangles: np.ndarray,
+    vertex: np.ndarray,
+    vec1: np.ndarray,
+    vec2: np.ndarray,
+    vec1_len: float,
+    vec2_len: float,
+    j: int,
+    cos_limit: float,
+    bevel: bool,
+) -> int:
+    """Set the centers, offsets, and triangles for a given path.
+
+    Parameters
+    ----------
+    centers : np.ndarray
+        Mx2 array central coordinates of path triangles.
+    offsets : np.ndarray
+        Mx2 array of the offsets to the central coordinates that need to
+        be scaled by the line width and then added to the centers to
+        generate the actual vertices of the triangulation
+    triangles : np.ndarray
+        (M-2)x3 array of the indices of the vertices that will form the
+        triangles of the triangulation
+    vertex : np.ndarray
+        The vertex of the path for which the centers,
+         offsets and triangles art calculated
+    vec1 : np.ndarray
+        The normal vector from previous vertex to the current vertex
+    vec2 : np.ndarray
+        The normal vector from the current vertex to the next vertex
+    vec1_len : float
+        The length of the vec1 vector, used for miter limit calculation.
+    vec2_len : float
+        The length of the vec2 vector, used for miter limit calculation.
+    j : int
+        The index of position to start putting items in the centers,
+        offsets and triangles arrays
+    cos_limit : float
+        Miter limit which determines when to switch from a miter join to a
+        bevel join
+    bevel : bool
+        Bool which if True causes a bevel join to always be used.
+        If False a bevel join will only be used when the miter limit is exceeded
+
+    Returns
+    -------
+    int
+        number of triangles, centers and offsets added to the arrays
+    """
+    centers[j] = vertex
+    centers[j + 1] = vertex
+    cos_angle = vec1[0] * vec2[0] + vec1[1] * vec2[1]
+    sin_angle = vec1[0] * vec2[1] - vec1[1] * vec2[0]
+    if sin_angle == 0:
```

Comment:
```suggestion

    if sin_angle == 0:
```

Add newline to create a more readable block.